### PR TITLE
Run lintian on Debian packages

### PR DESCRIPTION
**Pull request by legoktm**
## Status

Ready for review

## Description

lintian is the Debian package linter and catches a number of things we can improve in our packages. For now I've suppressed everything and annotated them with TODO/FIXME for others to take on or address incrementally.

lintian is run at the end of `make build-debs` unless the "FAST" environment variable is set. Any warning or error causes the build to fail, but it can be suppressed in the corresponding lintian-overrides file.

Internally there's a bit of work so we can identify the *.changes file that we just built, so we first build into a temporary directory, copy the files into our repository's build, then run lintian in the temporary directory.

Refs #1785.

## Test Plan

* [x] `make build-debs` passes locally
* [x] `DEBIAN_VERSION=bookworm make build-debs` passes locally too
* [x] `FAST=1 make build-debs` doesn't run lintian
* [x] CI passes
* [x] Infra moving the "Required" label from the now-renamed build-debs2/bullseye job to lintian/bullseye.

### CODE REVIEW

File: debian/securedrop-client.lintian-overrides
Comment (cfm) [inline debian/securedrop-client.lintian-overrides:2]: Nit: This file takes a moment to scan, because it looks like it's—

```
directive
# comment
```

—but it's actually—

```
# comment
directive
```

—after the first directive, which isn't commented.  Maybe add an extra newline before each directive?

```directive

# comment
directive

# comment
directive

# ...
```

File: scripts/build-debs.sh
Comment (cfm) [inline scripts/build-debs.sh:66]: Typo:

```suggestion
    # We suppress mismatched-override because of subtle differences in
```

**Review by cfm — CHANGES_REQUESTED** [review]
* [ ] `make build-debs` passes locally

Failed with `source-is-missing` until I deleted:
* `./client/htmlcov/`; and
* `./target/`.

We can't realistically enforce `make clean` before a local `make build-debs`.  Can we further ignore these transient paths that are generated by other tools?

* [x] `DEBIAN_VERSION=bookworm make build-debs` passes locally too
* [x] `FAST=1 make build-debs` doesn't run lintian
* [x] CI passes
* [ ] Infra moving the "Required" label from the now-renamed build-debs2/bullseye job to lintian/bullseye.

I will coordinate this in `#infrastructure` after approval and before merge.

**Comment by legoktm** [comment]
> We can't realistically enforce `make clean` before a local `make build-debs`. Can we further ignore these transient paths that are generated by other tools?

Yes, we can exclude those two paths from the package itself (ideally we'd just do something like `git clean -ffdx` but that would take away our ability to test uncommitted changes in the packaging workflow, which I think would be more irritating).

File: scripts/build-debs.sh
Comment (legoktm) [reply scripts/build-debs.sh:66]: in fact, this comment is outdated because previously I had it try to exclude the `mismatched-override` tag but removed it later, so dropped this whole comment

File: debian/securedrop-client.lintian-overrides
Comment (legoktm) [reply debian/securedrop-client.lintian-overrides:2]: Done!

**Comment by legoktm** [comment]
OK, files have now been excluded via d/source/options.

**Review by cfm — APPROVED** [review]
Great.  Thanks, @legoktm.  Approving now; will merge pending consultation in `#infrastructure`.

**Pull request merged**
